### PR TITLE
Adds get and delete methods for collections

### PR DESCRIPTION
**Pull request by GunnarHolwerda**

### CODE REVIEW

File: keen/client.py [hunk 1]
```diff
@@ -492,6 +492,42 @@ def funnel(self, steps, timeframe=None, timezone=None, max_age=None, all_keys=Fa
...
+    def delete_collection(self, collection):
+        """ Performs a deletion of a collection
+
+
+        :param collection_name: the name of the collection
+        """
+
+        self.api.delete_collection(collection)
+
+    def delete_collections(self, collections):
```
Comment (stroud109) [inline keen/client.py]: I don't think we need this function. That said, I'd be curious to hear why you think it's important. Could you tell me about your use case?

File: keen/api.py [hunk 2]
```diff
@@ -140,6 +140,27 @@ def post_events(self, events):
...
 
+    def delete_collection(self, collection):
+        """
+        Deletes a single collection in the Keen IO API. The master key must be set first.
+
+        :param event: a Collection to delete
+        """
+        if not self.master_key:
+            raise exceptions.InvalidEnvironmentError(
+                "The Keen IO API requires a master key to delete events. "
```
Comment (stroud109) [inline keen/api.py]: Could you update the language to say something like "delete a collection"?

File: keen/api.py [hunk 3]
```diff
@@ -190,6 +211,45 @@ def delete_events(self, event_collection, params):
         self.error_handling(response)
         return True
 
+    def get_collection(self, collection_name):
+        """
+        Extracts info about a collection using the Keen IO API. A master key must be set first.
```
Comment (stroud109) [inline keen/api.py]: How about something like "Returns event collection schema.."?

File: keen/api.py [hunk 4]
```diff
@@ -190,6 +211,45 @@ def delete_events(self, event_collection, params):
...
+
+        :param event_name: the name of the collection to retrieve info for
```
Comment (stroud109) [inline keen/api.py]: Could you change "event_name" to "event_collection" so it's consistent with the rest of the doc strings?

File: keen/api.py [hunk 5]
```diff
@@ -190,6 +211,45 @@ def delete_events(self, event_collection, params):
...
+        """
+        if not self.master_key:
+            raise exceptions.InvalidEnvironmentError(
+                "The Keen IO API requires a master key to get events. "
+                "Please set a 'master_key' when initializing the "
+                "KeenApi object."
+            )
+        url = "{0}/{1}/projects/{2}/events/{3}".format(self.base_url, self.api_version,
+                                                       self.project_id, collection_name)
```
Comment (stroud109) [inline keen/api.py]: `collection_name` (also on line 214) should be `event_collection` for the sake of consistency.

File: keen/client.py [hunk 6]
```diff
@@ -492,6 +492,42 @@ def funnel(self, steps, timeframe=None, timezone=None, max_age=None, all_keys=Fa
 
         return self.api.query("funnel", params, all_keys=all_keys)
 
+    def get_collection(self, collection_name):
+        """
+
+        Returns a description of a collection
+
+        :param event_name: the name of the event to get the collection info from
```
Comment (stroud109) [inline keen/client.py]: (Same request as the one on [api.py:118](https://github.com/keenlabs/KeenClient-Python/pull/73/files#diff-31115e53b9da1a9b8ce9f337f6c5285fR218) )

File: keen/api.py [hunk 7]
```diff
@@ -190,6 +211,45 @@ def delete_events(self, event_collection, params):
...
+        headers = {"Authorization": self.master_key}
+        response = self.fulfill(HTTPMethods.GET, url, headers=headers, timeout=self.get_timeout)
+        self.error_handling(response)
+
+        return response.json()
+
+    def get_all_collections(self):
+        """
+        Extracts info about all collections using the Keen IO API. A master key must be set first.
```
Comment (stroud109) [inline keen/api.py]: How about "Return schema information for all the event collections in a given project.." (I'm using this [page](https://keen.io/docs/api/?shell#inspect-all-collections) in Keen's Docs for reference.)

**Comment by stroud109** [comment]
This is looking great @GunnarHolwerda! Thanks so much! Could you add a few tests to go along with the new methods?

File: keen/client.py (hunk 1 above)
Comment (GunnarHolwerda) [inline keen/client.py]: I added this function before I fully understood how everything was working. I will go ahead and remove it. Thanks!

**Comment by GunnarHolwerda** [comment]
I have deleted all of the collection methods from client.py and left them to reside in api.py. Thank you so much for the feedback. I will add tests on Monday.

**Comment by GunnarHolwerda** [comment]
@stroud109 I have made your suggested changes. Do I still need the unit tests if the none of the collection methods I added are no longer on client.py? Thanks!

**Comment by stroud109** [comment]
Hey @GunnarHolwerda, yes adding a test for each of the new methods would be still be great! For reference, here's how I wrote the test for [delete_events](https://github.com/keenlabs/KeenClient-Python/blob/master/keen/tests/client_tests.py#L468). Let me know if there's any way I can help :smile:

**Comment by GunnarHolwerda** [comment]
Hey @stroud109 , I have added the unit tests. Let me know if there is anything else I can do! Thanks for all of your help!

**Comment by stroud109** [comment]
:+1: Thanks so much @GunnarHolwerda! I'll include these additions in the next update.
